exporter.py

```diff
@@ -0,0 +1,105 @@
+import logging
+import time
+from os import getenv
+
+import os
+
+import urllib3
+from prometheus_client import REGISTRY, start_http_server
+
+from collectors.PyVimServiceCollector import PyVimServiceCollector
+from collectors.SshServiceCollector import SshServiceCollector
+from collectors.EsxiOverallStateCollector import EsxiOnlineStateCollector
+from modules.Exceptions import VCenterException
+
+# init logger once
+logger = logger = logging.getLogger('esxi-exporter')
+logger.setLevel(logging.DEBUG)
+ch = logging.StreamHandler()
+formatter = logging.Formatter('[%(asctime)s] [%(levelname)s] %(message)s')
+ch.setFormatter(formatter)
+logger.addHandler(ch)
+
+# disable ssl warning
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+
+class Exporter:
+
+    def run_prometheus_server(self, port: int) -> None:
+        """
+        Starts a prometheus server which will stay running
+        :param port: The port the server should listen on
+        """
+
+        # startup prometheus server
+        logger.info("starting http server...")
+        start_http_server(port)
+
+        # register collectors
+        if not bool(getenv('DISABLE_PYVIM', False)):
+            logger.info("registering collector: PyVimServiceCollector... ")
+            REGISTRY.register(PyVimServiceCollector())
+        if not bool(getenv('DISABLE_SSH', False)):
+            logger.info("registering collector: sshServiceCollector...")
+            REGISTRY.register(SshServiceCollector())
+        if not bool(getenv('DISABLE_OVERALLSTATE', False)):
+            logger.info("registering collector: EsxiOverallStateCollector...")
+            REGISTRY.register(EsxiOnlineStateCollector())
+
+        logger.info("exporter is ready")
+
+        # stay alive
+        while True:
+            time.sleep(1)
+
+    def run(self):
+        try:
+            self.run_prometheus_server(int(getenv('EXPORTER_PORT', 1234)))
+        except Exception as ex:
+            logger.critical(ex)
+            raise ex
+
+    def check_config(self):

```

Comment:
1. You are checking environment vars, not a config. However, I would completely change this.

2. I would really encourage you, to remove the amount of ENV vars to a bare minimum, where it makes sense (PW, vcenter url etc). Your options (workercount, cachetime, blacklist etc pp) should really go into a config file which marks the default.

Reply:
done.